sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -57,6 +58,18 @@ internal set
             }
         }
 
+        internal static bool TryGetInternalInstance([NotNullWhen(true)] out IDeploymentInternal? instance)
+        {
+            if (_instance.Value != null)
```

Comment:
Hmm, I think it's okay. If _instance was shared across threads here, we'd need to avoid races by reading it only once:

```
value = _instance.Value
```

However maybe if it's AsyncLocal I think it's race-proof as it is. Anyway there is super low contention here.